src/transformStateWithClones.js

```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        currentState = { ...currentState };
```

Comment:
[DONT REPEAT YOURSELF]: The creation of a new object copy for the `currentState` at the beginning of the 'removeProperties' case is unnecessary because the `currentState` is already a separate object from the original `state`. You can remove this line to avoid redundant object copying.